src/test/java/com/synopsys/integration/jenkins/detect/service/DetectArgumentServiceTest.java

```diff
@@ -0,0 +1,203 @@
+
+        // These are the automatically appended detect properties added by DetectArgumentService(). Tests will update if needed.
+        // They should NOT be visible in the output log message but should be within the return List
+        expectedHiddenDetectProperties = new LinkedHashMap<>();
+        expectedHiddenDetectProperties.put(jenkinsVersionParam, expectedJenkinsVersion);
+        expectedHiddenDetectProperties.put(pluginVersionParam, expectedJenkinsPluginVersion);
+
+        // Create object in test
+        detectArgumentService = new DetectArgumentService(jenkinsIntLogger, jenkinsVersionHelper);
+    }
+
+    private String createDetectPropertiesInputString() {
+        return inputDetectProperties.keySet().stream()
+                   .map(key -> key + "=" + inputDetectProperties.get(key))
+                   .collect(Collectors.joining(" "));
+    }
+
+    @Test
+    public void testGetDetectArguments() {
+        List<String> detectCommandLine = detectArgumentService.getDetectArguments(intEnvironmentVariables, strategyEscaper, invocationParameters, createDetectPropertiesInputString());
+
+        assertEquals(6, detectCommandLine.size()); // Invocation args (1) + passed args (2) + auto added args (3)
+        commonValidation(detectCommandLine, expectedVisibleDetectProperties, expectedHiddenDetectProperties);
+    }
+
+    @Test
+    public void testShouldEscapeFalse() {
+        intEnvironmentVariables.put("DETECT_PLUGIN_ESCAPING", "false");
+
+        List<String> detectCommandLine = detectArgumentService.getDetectArguments(intEnvironmentVariables, strategyEscaper, invocationParameters, createDetectPropertiesInputString());
+
+        assertEquals(6, detectCommandLine.size()); // Invocation args (1) + passed args (2) + auto added args (3)
+        commonValidation(detectCommandLine, expectedVisibleDetectProperties, expectedHiddenDetectProperties);
+    }
+
+    @Test
+    public void testVariableReplaced() {
+        // Set up intEnvironmentVariables to only contain custom variable to verify search and replace of the same
+        IntEnvironmentVariables intEnvironmentVariables = new IntEnvironmentVariables(false);
+        intEnvironmentVariables.put("TRUST_CERT", "false");
+        inputDetectProperties.put("--blackduck.trust.cert", "$TRUST_CERT");
+        expectedVisibleDetectProperties.put("--blackduck.trust.cert", "false");
+
+        List<String> detectCommandLine = detectArgumentService.getDetectArguments(intEnvironmentVariables, strategyEscaper, invocationParameters, createDetectPropertiesInputString());
+
+        assertEquals(7, detectCommandLine.size()); // Invocation args (1) + passed args (3) + auto added args (3)
+        commonValidation(detectCommandLine, expectedVisibleDetectProperties, expectedHiddenDetectProperties);
+    }
+
+    @Test
+    public void testVariableNotReplaced() {
+        // This is used to validate a log entry from calling handleVariableReplacement()
```

Comment:
I think this test does fine expressing its intent-- it probably doesn't need the comment here or on line 133.